silx/io/dictdump.py

```diff
@@ -291,78 +287,83 @@ def dicttoh5(treedict, h5file, h5path='/',
                     numpy.asarray(value).dtype.type == numpy.unicode_):
                 value = numpy.array(value, dtype=h5py.special_dtype(vlen=str))
 
-            h5f[h5path + key[0]].attrs[key[1]] = value
+            h5f[h5name].attrs[attr_name] = value
 
 
-def dicttonx(
-    treedict,
-    h5file,
-    h5path="/",
-    mode="w",
-    overwrite_data=False,
-    create_dataset_args=None,
-):
-    """
-    Write a nested dictionary to a HDF5 file, using string keys as member names.
-    The NeXus convention is used to identify attributes with ``"@"`` character,
-    therefor the dataset_names should not contain ``"@"``.
+def nexus_to_h5_dict(treedict, parents=tuple()):
```

Comment:
Shouldn't this be private?

Reply:
The functions `h5_to_nexus_dict` (returns a dict without special objects) and `nexus_to_h5_dict` (returns a dict with special objects) could be useful if someone want to serialize/deserialize?

Reply:
OK, if that can be useful let's keep it, from looking at the code, I though that was only for internal use.